hypha/apply/determinations/templates/determinations/includes/determination_button.html

```diff
@@ -1,6 +1,14 @@
 {% load determination_tags workflow_tags %}
 {% if request.user|show_determination_button:submission %}
-    <a target="_blank" href="{% url 'apply:submissions:determinations:form' submission_pk=submission.id %}" class="button button--primary button--full-width">
-        {% if submission.determination.is_draft %}Update draft{% else %}Add Determination{% endif %}
+    <a target="_blank" href="{% url 'apply:submissions:determinations:form' submission_pk=submission.id %}" class="button button--primary button--full-width {{ class }}">
+        {% if request.user|has_draft_determination:submission %}
```

Comment:
I'm not sure why you updated this. Determination should be a 1-1 with the submission - so the user is not relevant only that they can work on determinations for this submission.

Reply:
hmm, I was thrown a bit by the existing code around determinations. In determination_block.html there is a `{% for determination in object.determinations.all %}` that implies that there can be multiple determinations for a submission. In fact this only ever shows one determination because if a determination is added when a draft determination already exists, the draft determination is updated. Also this existing conditional doesn't work as `determination` doesn't exist on `submission` so the button always shows `Add Determination`